index.js

```diff
@@ -48,17 +49,26 @@ function requestbody(opts) {
     var body = {};
     // so don't parse the body in strict mode
     if (!opts.strict || ["GET", "HEAD", "DELETE"].indexOf(this.method.toUpperCase()) === -1) {
-      if (opts.json && this.is('json'))  {
-        body = yield buddy.json(this, {encoding: opts.encoding, limit: opts.jsonLimit});
-      }
-      else if (opts.urlencoded && this.is('urlencoded')) {
-        body = yield buddy.form(this, {encoding: opts.encoding, limit: opts.formLimit});
-      }
-      else if (opts.text && this.is('text')) {
-        body = yield buddy.text(this, {encoding: opts.encoding, limit: opts.textLimit});
-      }
-      else if (opts.multipart && this.is('multipart')) {
-        body = yield formy(this, opts.formidable);
+      try {
+        if (opts.json && this.is('json'))  {
+          body = yield buddy.json(this, {encoding: opts.encoding, limit: opts.jsonLimit});
+        }
+        else if (opts.urlencoded && this.is('urlencoded')) {
+          body = yield buddy.form(this, {encoding: opts.encoding, limit: opts.formLimit});
+        }
+        else if (opts.text && this.is('text')) {
+          body = yield buddy.text(this, {encoding: opts.encoding, limit: opts.textLimit});
+        }
+        else if (opts.multipart && this.is('multipart')) {
+          body = yield formy(this, opts.formidable);
+        }
+        
+      } catch(parsingError) {
+        if (typeof(opts.onError) == 'function') {
+          opts.onError(parsingError, this);
```

Comment:
Ugh, sorry. Looking back at the length of this, I'm going to agree with the rest of the commenters that it's best if we keep it as short as possible. The nice thing about this module is that it simply does a very few things, with everything on a single "page" of code.

To that effect, any 2 liner should work ... even simpler, some flavor of this?:

```js
      } catch(parsingError) {
         if (!opts.onEror) throw parsingError;
         opts.onError(parsingError, this);
      }
```

Opinions?